src/layouts/Layout.astro

```diff
@@ -14,35 +14,102 @@ const currentYear = Temporal.Now.plainDateISO().year;
 <html lang="en">
   <head>
     <meta charset="UTF-8" />
-    <meta name="description" content="Astro description" />
+    <meta name="description" content="All Bugs Are Fixed - A Developer Blog" />
     <meta name="viewport" content="width=device-width" />
     <link rel="icon" type="image/svg+xml" href="/favicon.svg" />
     <meta name="generator" content={Astro.generator} />
     <title>{title}</title>
-    <link rel="stylesheet" href="/src/styles/tailwind.css" />
     <link rel="stylesheet" href="/src/styles/global.css" />
     <script is:inline>
       import { Temporal, Intl, toTemporalInstant } from "@js-temporal/polyfill";
       globalThis.Temporal = Temporal;
       globalThis.Intl = Intl;
       Date.prototype.toTemporalInstant = toTemporalInstant;
     </script>
+    <script is:inline>
+      // Initialize theme to prevent flash of unstyled content
+      const theme = (() => {
+        if (typeof localStorage !== 'undefined' && localStorage.getItem('theme')) {
+          return localStorage.getItem('theme');
+        }
+        if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
+          return 'dark';
+        }
+        return 'light';
+      })();
+
+      if (theme === 'dark') {
+        document.documentElement.classList.add('dark');
+      } else {
+        document.documentElement.classList.remove('dark');
+      }
+    </script>
   </head>
-  <body class="bg-white dark:bg-gray-900 text-gray-900 dark:text-gray-100">
-    <div class="max-w-3xl mx-auto px-4 sm:px-6 lg:px-8">
-      <header class="flex justify-between items-center py-8">
-        <a href="/" class="text-2xl font-bold">All Bugs Are Fixed</a>
-        <nav class="flex items-center space-x-4">
-          <a href="/blog/" class="hover:underline">Blog</a>
+  <body class="bg-gray-50 dark:bg-[hsl(210deg,30%,8%)] text-gray-900 dark:text-gray-100 flex flex-col min-h-screen transition-colors duration-300">
+    <!-- Gradient Top Border -->
+    <div class="h-2 w-full bg-gradient-to-r from-josh-primary to-josh-secondary"></div>
+
+    <header class="w-full px-6 py-8">
+      <div class="max-w-[1100px] mx-auto flex justify-between items-center">
+        <a href="/" class="text-2xl font-bold bg-clip-text text-transparent bg-gradient-to-r from-josh-primary to-josh-secondary hover:opacity-80 transition-opacity">
+          All Bugs Are Fixed
+        </a>
+        <nav class="hidden md:flex items-center space-x-8 font-medium">
+          <a href="/blog/" class="hover:text-josh-primary transition-colors">Blog</a>
+          <a href="#" class="hover:text-josh-primary transition-colors">About</a>
           <ThemeToggle />
         </nav>
-      </header>
-      <main class="py-8">
+        <div class="md:hidden">
+          <ThemeToggle />
```

Comment:
I feel like an attribute of theme toggle here could be from the earlier lookup from local storage as a property provided to this